FocusOn/ViewModel/TodayViewModel.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  TodayView-ViewModel.swift
+//  FocusOn
+//
+//  Created by Alexandra Ivanova on 25/03/2022.
+//
+
+import Foundation
+
+class TodayViewModel: ObservableObject {
+    @Published var goal = Goal()
+
+    func checkGoalCompletionStatus() {
+        if (goal.completionStatus) {
+            goal.completionStatus = false
+            goal.tasks[0].completionStatus = false
+            goal.tasks[1].completionStatus = false
+            goal.tasks[2].completionStatus = false
+
+            if !(goal.tasks[0].completionStatus &&
+                 goal.tasks[1].completionStatus &&
+                 goal.tasks[2].completionStatus) {
+                goal.completionStatus = false
+            }
+        } else {
+            goal.completionStatus = true
+            goal.tasks[0].completionStatus = true
+            goal.tasks[1].completionStatus = true
+            goal.tasks[2].completionStatus = true
+        }
```

Comment:
This is what we call business rule, it belongs to the entity as the view model only makes sure data is displayable. In some architectures, the view models also fetch the data, this is your case at the moment.